src/Algebra/Polynomial/Base1.agda

```diff
@@ -0,0 +1,82 @@
+open import Algebra.Bundles using (Ring)
+open import Data.Nat.Base as ℕ using (ℕ; _⊔_; suc; pred)
+open import Data.Product.Base using (_,_)
+open import Data.Vec.Base as Vec using ([]; _∷_)
+import Level as L
+
+module Algebra.Polynomial.Base1
+  {ℓ₁ ℓ₂} (R : Ring ℓ₁ ℓ₂)
+  where
+
+open import Algebra.Polynomial.Poly.Base1 R as Poly
+  using (Poly; zeros)
+
+open Ring R
+  using (0#; 1#)
+  renaming (Carrier to A)
+
+private
+  variable
+    m n k : ℕ
+    p : Poly m
+    q : Poly n
+    r : Poly k
+
+---------------------------------------------------
+-- Types
+
+record Polynomial : Set ℓ₁ where
+  constructor _,_
+  field
+    degree : ℕ
+    polynomial : Poly degree
+
+private
+  variable
+    P Q W : Polynomial
+
+-- Equivalence relation for representations of polynomials
+infix 4 _≈_
+_≈_ : Polynomial → Polynomial → Set (ℓ₁ L.⊔ ℓ₂)
+(m , p) ≈ (n , q) = p Poly.≈ q
+
+refl : P ≈ P
+refl = Poly.refl
+
+sym : P ≈ Q → Q ≈ P
+sym = Poly.sym
+
+trans : P ≈ Q → Q ≈ W → P ≈ W
+trans = Poly.trans
+
+--------------------------------------------------
+-- Constants
+
+zero : Polynomial
+zero = (0 , [])
+
+one : Polynomial
+one = (1 , (1# ∷ []))
+
+---------------------------------------------------
+-- Operations
+
+-- Multiply the polynomial by a factor of x
```

Comment:
I'd say "by x" rather than "by a factor of x", as I wondering "what factor?" 

And why not implement `shift :: ℕ → Polynomial → Polynomial instread?